OpenTargets.pm

```diff
@@ -0,0 +1,248 @@
+}
+
+sub _get_colnames {
+  my $self = shift;
+
+  # Open file header
+  open IN, "tabix -H " . $self->{_files}[0] . " |"
+    or die "ERROR: cannot open tabix file for " . $self->{_files}[0];
+
+  # Get last line from header
+  my $last;
+  $last = $_ while <IN>;
+  $last =~ s/(^#|\n$)//g;
+  close IN;
+
+  # Parse column names from header
+  my @cols = split /\t/, $last;
+  @cols = splice @cols, 4; # first columns only identify the variant
+
+  # Prefix all column names
+  return _prefix_cols(\@cols);
+}
+
+sub _parse_colnames {
+  my $self = shift;
+  my $cols = shift;
+
+  # Parse file columns
+  $self->{colnames} = $self->_get_colnames();
+  
+  if ($cols eq "all") {
+    $self->{cols} = $self->{colnames};
+  } else {
+    my @cols = split(/:/, $cols);
+
+    # Prefix all column names
+    $self->{cols} = _prefix_cols(\@cols);
+
+    # Check validity of all columns
+    my @invalid_cols = grep { !($_ ~~ $self->{colnames}) } @{ $self->{cols} };
+    die "\n  ERROR: The following columns were not found in file header: ",
+      join(", ", @invalid_cols), "\n" if @invalid_cols;
+  }
+}
+
+sub new {
+  my $class = shift;  
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+  $self->get_user_params();
+
+  my $param_hash = $self->params_to_hash();
+  my $tr_match = $param_hash->{transcript_match};
+  $self->{transcript_match} = defined $tr_match ? $tr_match : 1;
+
+  my $aa_changes = $param_hash->{single_aminoacid_changes};
+  $self->{single_aa_changes} = defined $aa_changes ? $aa_changes : 1;
+
+  # Check file
+  my $file = $param_hash->{file};
+  die "\n  ERROR: No file specified\nTry using 'file=path/to/data.tsv.bz'\n"
+     unless defined($file);
+  $self->add_file($file);
+
+  # Parse column names
+  my $cols = $param_hash->{cols} || "l2g";
+  $self->_parse_colnames($cols);
+
+  return $self;
+}
+
+sub feature_types {
+  return ['Feature', 'Intergenic'];
+}
+
+sub get_header_info {
+  my $self = shift;
+  my %header;
+  my @keys = @{ $self->{colnames} };
+
+  my $description = "column from " . basename $self->{_files}[0];
+  my @vals = map { $description } @keys;
+  @header{ @keys } = @vals;
+
+  # Custom headers
+  $header{"OpenTargets_l2g"}  = "Locus-to-gene (L2G) scores to predict causal genes at GWAS loci; " . $description;
+
+  # Filter by user-selected columns
+  %header = map { $_ => $header{$_} } @{ $self->{cols} };
+
+  return \%header;
+}
+
+sub _filter_selected_cols {
+  my $self = shift;
+  my $res = shift;
+
+  my %tmp = %$res;
+  %tmp = map { $_ => $res->{$_} } @{ $self->{cols} };
+  return \%tmp;
+}
+
+sub run {
+  my ($self, $tva) = @_;
+  
+  my $vf = $tva->variation_feature;
+  my $allele = $tva->base_variation_feature->alt_alleles;
+  my @data = @{$self->get_data($vf->{chr}, $vf->{start} - 2, $vf->{end})};
+
+  foreach (@data) {
+    # Check if genes match
+    next unless 
+      $tva->transcript->{_gene_stable_id} eq $_->{result}->{OpenTargets_geneId};
```

Comment:
This line fails for intergenic variants:
```
WARNING: Plugin 'OpenTargets' went wrong: Can't locate object method "transcript" via package "Bio::EnsEMBL::Variation::IntergenicVariationAllele" at /hps/software/users/ensembl/repositories/dlemos/VEP_plugins/OpenTargets.pm line 193
```

Reply:
Now we don't match by gene, as expected, so this is not an issue anymore.